aptos-move/block-executor/src/executor.rs

```diff
@@ -699,6 +719,25 @@ where
                 true,
             );
 
+            let res = match res {
+                ExecutionStatus::Success(output) => {
+                    // Apply the writes/deltas to the versioned_data_cache.
+                    ExecutionStatus::Success(output)
+                },
+                ExecutionStatus::SkipRest(output) => {
+                    // Apply the writes/deltas and record status indicating skip.
+                    ExecutionStatus::SkipRest(output)
+                },
+                ExecutionStatus::Abort(err) => {
+                    // Record the status indicating abort.
+                    ExecutionStatus::Abort(Error::UserError(err))
```

Comment:
not sure i understand why we need this change, also it can be written like
```
a @ ExecutionStatus(_) | ExecutionStatus::SkipRest(_) => a,
```

Reply:
I tried that first but it confuses the compiler . This is a little verbose but its the only way I could get this working. We also need to do the same in parallel execution code path BTW.

```

719 |               let res = match res {
    |  _______________________-
720 | |                 status @ ExecutionStatus::Success(_) | ExecutionStatus::SkipRest(_) => status,
    | |                                                                                        ------ this is found to be of type `task::ExecutionStatus<<E as task::ExecutorTask>::Output, <E as task::ExecutorTask>::Error>`
721 | |                 ExecutionStatus::Abort(err) => {
722 | |                     // Record the status indicating abort.
723 | |                     ExecutionStatus::Abort(Error::UserError(err))
    | |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `ExecutionStatus<<... as ExecutorTask>::Output, ...>`, found `ExecutionStatus<_, Error<...>>`
724 | |                 },
725 | |             };
```